src/libdrakvuf/vmi.c

```diff
@@ -1208,14 +1208,43 @@ void drakvuf_loop(drakvuf_t drakvuf)
 
     while (!drakvuf->interrupted)
     {
-        //PRINT_DEBUG("Waiting for events in DRAKVUF...\n");
-        status_t status = vmi_events_listen(drakvuf->vmi, 1000);
-
-        if ( VMI_SUCCESS != status )
+        int rc = poll(drakvuf->event_fds, drakvuf->event_fd_cnt, 1000);
+        if (rc == 0)
+        {
+            continue;
+        }
+        else if (rc < 0)
         {
-            PRINT_DEBUG("Error waiting for events or timeout, quitting...\n");
+            perror("drakvuf_loop poll error: ");
             drakvuf->interrupted = -1;
         }
+
+        /* check and process each fd if it was raised */
+        for (int poll_ix=0; poll_ix<drakvuf->event_fd_cnt; poll_ix++)
+        {
+            if ( !(drakvuf->event_fds[poll_ix].revents & POLLIN) )
+            {
+                continue;
+            }
+
+            fd_info_t fd_info = &drakvuf->fd_info_lookup[poll_ix];
+
+            /* this is a normal VMI event */
+            if (fd_info->flags & EVENT_FD_VMI)
+            {
+                status_t status = vmi_events_listen(drakvuf->vmi, 1000);
```

Comment:
This should be `vmi_events_listen(drakvuf->vmi, 0);` otherwise libvmi itself will do a `poll` internally.

Reply:
Fixed!